File: languages/go/entity/identity/integer_attribute_descriptor.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// IntegerAttributeDescriptor represents a descriptor for an attribute with an integer value.
type IntegerAttributeDescriptor struct {
	AttributeDescriptor
}

// NewIntegerAttributeDescriptor creates a new IntegerAttributeDescriptor instance with the provided details.
func NewIntegerAttributeDescriptor(name string, id string, description string, required bool) IntegerAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return IntegerAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewNumberAttributeDescriptorFromProto(s *proto.IntegerAttributeDefinition) IntegerAttributeDescriptor {
	if s == nil {
		return IntegerAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return IntegerAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s IntegerAttributeDescriptor) ToProto() *proto.IntegerAttributeDefinition {
	return &proto.IntegerAttributeDefinition{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}

File: languages/js/src/entity/identity/integer_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/bloock_identity_entities";
import { AttributeDescriptor } from "./attribute_descriptor";

/**
 * Represents a descriptor for an attribute with an integer value.
 */
export class IntegerAttributeDescriptor extends AttributeDescriptor {
  public toProto(): identityEntitiesProto.IntegerAttributeDefinition {
    return identityEntitiesProto.IntegerAttributeDefinition.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.IntegerAttributeDefinition
  ): IntegerAttributeDescriptor {
    return new IntegerAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}
